fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -821,6 +821,24 @@ public boolean isPreviousTransactionsFinished(long endTransactionId, long dbId)
         }
         return true;
     }
+
+    // check if there exists a intersection between the source tableId list and target tableId list
+    // if one of them is null or empty, that means that we don't know related tables in tableList,
+    // we think the two lists may have intersection for right ordered txns
+    public boolean isIntersectionNotEmpty(List<Long> sourceTableIdList, List<Long> targetTableIdList) {
+        if (sourceTableIdList == null || sourceTableIdList.isEmpty() || targetTableIdList == null ||
+                targetTableIdList.isEmpty()) {
+            return true;
+        }
+        for (int i = 0; i < sourceTableIdList.size(); i++) {
```

Comment:
the method is low efficiency, use `sourceTableIdList.retainAll(targetTableIdList);sourceTableIdList.isEmpty()` instread?

Reply:
> the method is low efficiency, use `sourceTableIdList.retainAll(targetTableIdList);sourceTableIdList.isEmpty()` instread?

in most case the table list size is 1, and we should not modify the table list content